### ISSUE TITLE
Switch to leveraging `pipenv`'s new `pipenv upgrade` command

### ISSUE THREAD

**Issue by jeffwidman** [dependabot/dependabot-core#6836]
> I have another update for dependabot team though -- which is in the `main` branch of `pipenv` is a a new `pipenv upgrade` command that should be perfect for dependabot's use case of upgrading the Pipfile/Pipfile.lock (doesn't require package install) -- and doesn't affect the whole set of lock file dependencies.  
> 
> Since `--selective-upgrade` and `--keep-outdated` are deprecated for removal, it would be useful to try updating dependabot to use this new command (it will be in the next release, maybe in a week but you can try it out from the main branch today).
> 
> If you are experiencing issues related to the upgrade path, feel free to open new issue reports and we will prioritize looking into them.   Thank you!

_Originally posted by @matteius in https://github.com/pypa/pipenv/issues/2791#issuecomment-1468694507_

**Comment by jeffwidman** [comment]
Unfortunately, bumping to the latest version of `pipenv` is currently blocked on us dropping support for Python 3.6:
* https://github.com/dependabot/dependabot-core/pull/6104#issuecomment-1397616478

**Comment by jeffwidman** [comment]
Upstream PR adding the feature:
* https://github.com/pypa/pipenv/pull/5617

Many thanks @matteius for adding this to help out the Dependabot / package updater bot use case. 😍
